src/Slim/Middleware/Session.php

```diff
@@ -83,48 +83,52 @@ public function __invoke(Request $request, Response $response, callable $next)
      */
     protected function startSession()
     {
-        $settings = $this->settings;
-        $name = $settings['name'];
+        $inactive = session_status() === PHP_SESSION_ACTIVE;
 
-        session_set_cookie_params(
-            $settings['lifetime'],
-            $settings['path'],
-            $settings['domain'],
-            $settings['secure'],
-            $settings['httponly']
-        );
+        if (!$inactive) {
+            $settings = $this->settings;
+            $name = $settings['name'];
 
-        $inactive = session_status() === PHP_SESSION_NONE;
+            session_set_cookie_params(
+                $settings['lifetime'],
+                $settings['path'],
+                $settings['domain'],
+                $settings['secure'],
+                $settings['httponly']
+            );
 
-        if ($inactive) {
+            $inactive = session_status() === PHP_SESSION_NONE;
```

Comment:
Why to define `$inactive` again if it's already defined before (method start)?
Maybe you should find a standard way to use `session_status` to get session active status. (also taking in account for disabled sessions)